server/Dockerfile

```diff
@@ -47,82 +50,93 @@ FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim" as php-build
+
+        cd /var/www/MISP || exit; git submodule update --init --recursive .
+
+        # Until MISP project specifies its required PyMISP version, we grab this from the submodule
+        pymispversion=$(head -n1 PyMISP/pymisp/__init__.py | awk '{print $3}' | tr -d \')
+
+        if [ -n "$pymispversion" ]; then
+            sed -i "s/pymisp$/pymisp==$pymispversion/" requirements.txt
+            cat requirements.txt
+        fi
+        # End of pymisp version fixing hack
+
+        # Remove modules only used during MISP build
+        set -- 'coveralls' 'codecov' 'requests-mock' 'nose' 'pip'
+        for mod in "$@"; do
+            sed "/${mod}/d" -i /var/www/MISP/requirements.txt
+        done;
+
+        # Add additional dependencies (container specific)
+        # The "set" line contains the list of modules we want to ensure are present.
+        # PYPI_MODULE_NAME_VERSION env vars can be set to specify the version desired,
+        # e.g. PYPI_SURICATA_VERSION="==2.0" to specify exactly version 2.0 for the suricata package
+        #
+        # 1. Check for presence of each module in requirements.txt
+        # 2. If missing, add it (with optional version from env (defaults to empty string))
+        # 3. If present, replace with our specified version if it exists, otherwise leave
+        #    the upstream version alone.
+        set -- "redis" "lief" "pydeep2" "python-magic" "misp-lib-stix2" "maec" "mixbox" "cybox"
+        for mod in "$@"; do
+            mod_version_var=$(echo "PYPI_${mod}_VERSION" | tr '[:lower:]' '[:upper:]' | tr '-' '_')
+            mod_version=$(eval "echo \"\$$mod_version_var\"")
+            grep -q ${mod} /var/www/MISP/requirements.txt
+            exists=$?
+            if [ "${exists}" -eq "1" ]; then
+                echo "Adding missing module ${mod} with version '${mod_version}'"
+                echo ${mod}${mod_version} >> /var/www/MISP/requirements.txt
+            else
+                if [ "$(echo ${mod_version} | wc -m)" -gt 1 ]; then
+                    echo "Overwriting existing module ${mod}, version '${mod_version}'"
+                    sed -i "/${mod}/s/.*/${mod}${mod_version}/" /var/www/MISP/requirements.txt
+                else
+                    echo "Skipping overwriting ${mod} due to missing version variable"
+                fi
+            fi
+        done;
+
+        pip3 wheel --no-cache-dir -w /wheels/ -r /var/www/MISP/requirements.txt
+        
+        # Remove files we do not care for
+        rm -r /var/www/MISP/PyMISP
+        find /var/www/MISP/INSTALL/* ! -name 'MYSQL.sql' -type f -exec rm {} +
+        find /var/www/MISP/INSTALL/* ! -name 'MYSQL.sql' -type l -exec rm {} +
+        # Remove most files in .git - we do not use git functionality in docker
+        find /var/www/MISP/.git/* ! -name HEAD -exec rm -rf {} +
+        # Diagnostics wants this file to be present and writable. Entrypoint will override any perm we set here
+        touch /var/www/MISP/.git/ORIG_HEAD
```

Comment:
Are lines 130 and 132 still needed after the merge?

Reply:
Yes, there is still a direct read of the HEAD file that causes an error in the log unless HEAD is present, while ORIG_HEAD is still expected to be present and writeable. I removed the part of my now merged upstream PR that excluded this file to increase the chances of getting merged (as the removal felt like it unnecessarily cluttered their code)